### PR TITLE
cmd: use CI_API_V4_URL env as gitlab API URL

### LINKED ISSUE #563: Make use of predefined CI_API_V4_URL instead of GITLAB_API
Since Gitlab 11.7, a variable containing the URL to the gitlab API v4 is automatically set by gitlab: [`CI_API_V4_URL`](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html#variables-reference).

reviewdog could use this instead of `GITLAB_API`, perhaps as a fall-back mechanism (try `GITLAB_API` first, fall back to `CI_API_V4_URL` if the former is not set) to not break existing configurations.

I have not checked the code but the change seems pretty straightforward, so I could probably work on a PR for this if you are interested.

### PR DESCRIPTION
**Pull request by roobre**
This should be a simple change, although it has not been extensively tested.

Closes #563.

### CODE REVIEW

File: cmd/reviewdog/main.go [hunk 1]
```diff
@@ -596,10 +597,19 @@ func gitlabClient(token string) (*gitlab.Client, error) {
 const defaultGitLabAPI = "https://gitlab.com/api/v4"
 
 func gitlabBaseURL() (*url.URL, error) {
-	baseURL := os.Getenv("GITLAB_API")
-	if baseURL == "" {
+	gitlabApi := os.Getenv("GITLAB_API")
+	gitlabV4Url := os.Getenv("CI_API_V4_URL")
```
Comment (haya14busa) [inline cmd/reviewdog/main.go:601]: `gitlabV4URL`.

Also, please fix the above golint error as well.
> `var gitlabApi should be gitlabAPI`

File: cmd/reviewdog/main.go [hunk 2]
```diff
@@ -596,10 +597,19 @@ func gitlabClient(token string) (*gitlab.Client, error) {
...
+
+	var baseURL string
+	switch true {
```
Comment (haya14busa) [inline cmd/reviewdog/main.go:604]: Can you use if-else clause? It's interesting, but simple if-else should be better.

Alternatively, you can create another function which returns URL string and use early return.

**Comment by haya14busa** [comment]
Also, can you update Unreleased section in the changelog as well? https://github.com/reviewdog/reviewdog/blob/master/CHANGELOG.md

File: cmd/reviewdog/main.go (hunk 1 above)
Comment (roobre) [reply cmd/reviewdog/main.go]: Done

File: cmd/reviewdog/main.go (hunk 2 above)
Comment (roobre) [reply cmd/reviewdog/main.go]: I thought about the function first, but then it seemed weird to me to have the current gitlabBaseURL to just be a wrapper for `url.Parse`. I have switched to the if-else chain.

**Pull request merged**